packages/ckeditor5-image/src/imageresize/imageresizebuttons.ts

```diff
@@ -179,8 +193,12 @@ export default class ImageResizeButtons extends Plugin {
 
 			// Execute command when an item from the dropdown is selected.
 			this.listenTo( dropdownView, 'execute', evt => {
-				editor.execute( ( evt.source as any ).commandName, { width: ( evt.source as any ).commandValue } );
-				editor.editing.view.focus();
+				if ( 'onClick' in evt.source ) {
+					( evt.source as any ).onClick();
```

Comment:
I hate these `ViewModels` and `as any` that it brings in so many places across the project :/ I wish we got rid of them. 

Maybe instead of duck typing we could assign a legitimate property to the view model like `#buttonType` or something and check that directly here? The outcome would be the same but this would look less like a magic.

Reply:
Maybe this is a more elegant way of doing things (even though we still have to cast):

```diff
diff --git a/packages/ckeditor5-image/src/imageresize/imageresizebuttons.ts b/packages/ckeditor5-image/src/imageresize/imageresizebuttons.ts
index 1775a9b136..7b25bb203a 100644
--- a/packages/ckeditor5-image/src/imageresize/imageresizebuttons.ts
+++ b/packages/ckeditor5-image/src/imageresize/imageresizebuttons.ts
@@ -193,10 +193,12 @@ export default class ImageResizeButtons extends Plugin {
 
 			// Execute command when an item from the dropdown is selected.
 			this.listenTo( dropdownView, 'execute', evt => {
-				if ( 'onClick' in evt.source ) {
-					( evt.source as any ).onClick();
+				const eventSource = evt.source as ImageResizeButtonModel;
+
+				if ( eventSource.type == 'custom' ) {
+					this.editor.plugins.get( 'ImageCustomResizeUI' )._showForm( this._resizeUnit );
 				} else {
-					editor.execute( ( evt.source as any ).commandName, { width: ( evt.source as any ).commandValue } );
+					editor.execute( eventSource.commandName, { width: eventSource.commandValue } );
 					editor.editing.view.focus();
 				}
 			} );
@@ -252,7 +254,6 @@ export default class ImageResizeButtons extends Plugin {
 		options: Array<ImageResizeOption>,
 		command: ResizeImageCommand
 	): Collection<ListDropdownItemDefinition> {
-		const customResizeUI = this.editor.plugins.get( 'ImageCustomResizeUI' );
 		const itemDefinitions = new Collection<ListDropdownItemDefinition>();
 
 		const optionsWithSerializedValues = options.map( option => {
@@ -282,14 +283,12 @@ export default class ImageResizeButtons extends Plugin {
 			if ( isCustomImageResizeOption( option ) ) {
 				definition = {
 					type: 'button',
-					model: new ViewModel( {
+					model: new ImageResizeButtonModel( {
 						label: this._getOptionLabelValue( option ),
 						role: 'menuitemradio',
 						withText: true,
 						icon: null,
-						onClick: () => {
-							customResizeUI._showForm( this._resizeUnit );
-						}
+						type: 'custom'
 					} )
 				};
 
@@ -299,13 +298,14 @@ export default class ImageResizeButtons extends Plugin {
 			} else {
 				definition = {
 					type: 'button',
-					model: new ViewModel( {
+					model: new ImageResizeButtonModel( {
 						commandName: 'resizeImage',
 						commandValue: option.valueWithUnits,
 						label: this._getOptionLabelValue( option ),
 						role: 'menuitemradio',
 						withText: true,
-						icon: null
+						icon: null,
+						type: 'fixed'
 					} )
 				};
 
@@ -349,3 +349,14 @@ function getIsOnCustomButtonCallback( allDropdownValues: Array<string | null> ):
 		dropdownValue => getIsOnButtonCallback( dropdownValue )( commandValue )
 	);
 }
+
+class ImageResizeButtonModel extends ViewModel {
+	declare public type: 'fixed' | 'custom';
+	declare public commandName: string;
+	declare public commandValue: string;
+	declare public icon?: string;
+	declare public label: string;
+	declare public onClick?: Function;
+	declare public role: string;
+	declare public withText: boolean;
+}
```